components/discord_bot/sources/new-guild-member/new-guild-member.mjs

```diff
@@ -0,0 +1,67 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  key: "discord_bot-new-guild-member",
+  name: "New Guild Member",
+  description: "Emit new event for every member added to a guild",
+  type: "action",
+  version: "0.0.1",
+  props: {
+    ...common.props,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  methods: {
+    ...common.methods,
+  },
+  async run({ $ }) {
+    const guildMembers = this._getGuildMemberIDs();
+    const { guildId } = this;
+    const params = {
+      limit: 100,
+    };
+    let anyNewMember = false;
+
+    while (true) {
```

Comment:
I don't get why you need a `while(true)` here.  Could you please explain to me?

Reply:
> I don't get why you need a while(true) here. Could you please explain to me?

It's due to pagination. When [listing guild members](https://discord.com/developers/docs/resources/guild#list-guild-members), the `after` param is **the highest user id in the previous page, inclusive**. So the only way to know if the final member was reached is if the response contains only one member.

Reply:
What if we store this after to not repeat the request with the previous users, is it possible? Maybe you can use a `do/while` until `after` to be `null`? Normally the `after` points to the last record on the page. So you can always save the last record as `after`. Does it make sense for you?

Reply:
Yeah, I thought about that but members with a lower `id` can be added to a guild after the run, so I think in those cases they would be skipped